### DISCUSSION TITLE
Form is using formProvider by default and examples passes control as props.

### DISCUSSION THREAD

**Discussion by hadson172** [shadcn-ui/ui#1212] [General]
Hi, I started to looking onto Form component from shadcn. It seems that it uses form provider by default, which could have serious performance implications :
https://www.react-hook-form.com/advanced-usage/#FormProviderPerformance

Additionaly, all examples in example section are using inputs by passing Control which is "optimisation" for FormProvider.  (directly passing control).

Am I missing something in here?

**Comment by simbakowo** [comment]
@shadcn ?
